dirty_equals/_other.py

```diff
@@ -1,7 +1,9 @@
 import json
-from typing import Any, Callable, TypeVar, overload
+from typing import Any, Callable, Set, TypeVar, overload
 from uuid import UUID
 
+from pydantic import AmqpDsn, AnyHttpUrl, AnyUrl, BaseModel, FileUrl, HttpUrl, PostgresDsn, RedisDsn
```

Comment:
I don't want to make pydantic a required dependency. Better to import in `IsUrl.__init__` and raise in `ImportError` with instructions on installing `pydantic` if it's not installed.

A bit like [this](https://github.com/pydantic/pydantic/blob/f8575aa3793c565cf006979c60ad62312406d5b5/pydantic/env_settings.py#L328)